subprojects/launcher/src/main/java/org/gradle/launcher/daemon/server/DaemonServices.java

```diff
@@ -142,6 +143,7 @@ protected GarbageCollectorMonitoringStrategy createGarbageCollectorMonitoringStr
         DaemonDiagnostics daemonDiagnostics = new DaemonDiagnostics(daemonLog, daemonContext.getPid());
         return ImmutableList.of(
             new HandleStop(get(ListenerManager.class)),
+            new HandleInvalidateLocations(this),
```

Comment:
This should match the name of the command I think.

Reply:
Do we have to pass `this`? That's always a smell...

Reply:
I replaced `this` by `get(GradleUserHomeScopeServiceRegistry)`.